### PR TITLE
Update `image_filter_test` and `color_filter_test.dart` for Material 3

### PR DESCRIPTION
**Pull request by TahaTesser**
Updated unit tests for `ColorFiltered` and `ImageFiltered` to have M2 and M3 versions.

More info in #139076

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks for continuing to add more tests!!

File: packages/flutter/test/widgets/image_filter_test.dart [hunk 1]
```diff
@@ -116,11 +116,49 @@ void main() {
...
       find.byType(ImageFiltered),
-      matchesGoldenFile('image_filter_matrix.png'),
+      matchesGoldenFile('image_filter_matrix_m2.png'),
     );
   });
 
-  testWidgets('Image filter - matrix with offset', (WidgetTester tester) async {
+  testWidgets('Image filter - matrix', (WidgetTester tester) async {
+    final ImageFilter matrix = ImageFilter.matrix(Float64List.fromList(<double>[
+      0.5, 0.0, 0.0, 0.0, //
```
Comment (QuncCccccc) [inline packages/flutter/test/widgets/image_filter_test.dart:125]: I'm actually not very familiar with ColorFilter and ImageFiler, why do we end each line with a "//"?

**Comment by jonahwilliams** [comment]
When this landed it caused skia gold to begin reporting untriaged goldens. I went ahead and accepted them, so things should green up.

The problem I believe was the force push will cause the skia gold check to not fire correctly. Just an FYI to try avoid force pushes if you're updating goldens.

**Comment by TahaTesser** [comment]
@jonahwilliams 
Thanks for the update! I usually push empty commit when this happens and golden testing fails. I'll remember to avoid force push when updating Goldens.

**Comment by jonahwilliams** [comment]
Looks like there are more web changes. I'm unsure if these are expected: https://flutter-gold.skia.org/detail?grouping=name%3Dwidgets.image_filter_matrix_offset_m3%26source_type%3Dflutter&digest=725f13536274b8ee715e64df3f2d6ddc

**Comment by TahaTesser** [comment]
> Looks like there are more web changes. I'm unsure if these are expected: https://flutter-gold.skia.org/detail?grouping=name%3Dwidgets.image_filter_matrix_offset_m3%26source_type%3Dflutter&digest=725f13536274b8ee715e64df3f2d6ddc

Could it be due to different web renderer? 
Screenshot 2024-11-25 at 21 12 43

**Comment by QuncCccccc** [comment]
Should we revert this for now to unblock the tree?

**Pull request merged**